docs/support-bundle-guide.md

```diff
@@ -0,0 +1,247 @@
+    namespace: kube-system
+```
+
+## The SupportBundleCollection CRD
+
+SupportBundleCollection CR is introduced to supplement the `antctl` command with
+three reinforcements:
+
+1. Allow users to collect support bundle files on ExternalNodes.
+2. Upload all the support bundle files into a user-provided SFTP Server.
+3. Support tracking status of a SupportBundleCollection CR.
+
+Details are provided in the next section.
+
+## Use case example
+
+### Running antctl commands
+
+Please refer to the corresponding [antctl user guide section](antctl.md#collecting-support-information).
+Note: `antctl supportbundle` only works on collecting the support bundles from Antrea Controller and
+Antrea Agent which is running on K8s Node, it does not work for the Agent on an external Node.
+
+### Applying SupportBundleCollection CRD
+
+In this section, we will create two SupportBundleCollections for Kubernetes Nodes and ExternalNodes.
+Assume we have a cluster with Nodes named "kube-worker" and "kube-master". In addition, we have
+set up an ExternalNode as `vm` in the `vm-ns` Namespace by following the instruction
+in [Install ExternalNode on VM](external-node.md#install-antrea-agent-on-vm).
+
+A Secret needs to be created in advance with the username and password of the SFTP Server.
+The Secret will be used in `authSecret` in the following YAML examples.
+
+```bash
+kubectl create secret generic support-bundle-secret  --from-literal=username='your-sftp-username'  --from-literal=password='your-sftp-password'
+```
+
+Then we could apply the following YAML files. The first one is to collect support bundle
+on Kubernetes Nodes "kube-worker" and "kube-master". The second one is to collect support
+bundle on ExternalNode "vm".
+
+```bash
+cat << EOF | kubectl apply -f -
+kind: SupportBundleCollection
+metadata:
+  name: collect-support-bundle-for-kube-nodes
+spec:
+  nodes:
+    nodeNames: # Empty list means all the Nodes
+      - kube-master
+      - kube-worker
+    matchLabels:
+      app: nginx   # Collect support bundle files from the Nodes which have been labeled as "app: nginx".
+  expirationMinutes: 10 # expirationMinutes is the requested duration of validity of the SupportBundleCollection. A SupportBundleCollection will be marked as Failed if it does not finish before expiration.
+  sinceTime: 2h # Collect the logs in the latest 2 hours. Collect all Agent logs if it is not set.
+  fileServer:
+    url: sftp://yourtestdomain.com:22/root/test
+  authentication:
+    authType: "BasicAuthentication"
+    authSecret:
+      name: support-bundle-secret
+      namespace: default # antrea-controller has been given the permission to read secrets in "default" namespace. 
+EOF
+```
+
+```bash
+cat << EOF | kubectl apply -f -
+apiVersion: crd.antrea.io/v1alpha1
+kind: SupportBundleCollection
+metadata:
+  name: collect-support-bundle-for-vm-external-node
+spec:
+  externalNodes:
+    nodeNames: # Empty list means all the ExternalNodes in the Namespace
```

Comment:
It may be ok to use `ExternalNode` in such cases, which can refer to the ExternalNode CR